src/classes/RecipeCollection.js

```diff
@@ -0,0 +1,48 @@
+class RecipeCollection {
+  constructor(allRecipes, allIngredients) {
+    this.recipes = allRecipes;
+    this.ingredients = allIngredients;
+  }
+
+  filterRecipe(searchInput) {
+    const recipeMatches = this.recipes.reduce((searchedRecipes, recipe) => {
+      if (recipe.name.toLowerCase().includes(searchInput.toLowerCase()) ||
+      recipe.tags.includes(searchInput.toLowerCase()) ||
+      recipe.ingredients.includes(searchInput.toLowerCase())) {
+        searchedRecipes.push(recipe);
+      }
+      return searchedRecipes;
+    }, [])
+    return recipeMatches;
+  }
+
+//Finding the ingredient in ingredientData that matches the ingredient input
+  findRecipe(ingredientInput) {
+    const ingredientMatch = this.ingredients.find(ingredient => {
+     if(ingredient.name === ingredientInput.toLowerCase()){
+       return ingredient
+     }
+     });
+    console.log(ingredientMatch)
+//im getting an error that recipes.ingredients.id.includes is not a function -- not sure if Im accessing this wrong??
+    const recipesWithIngredient = this.recipes.filter(recipe => {
+      return recipe.ingredients.id.includes(ingredientMatch.id)
+      });
```

Comment:
I think you are getting that error because you are trying to do the .includes method on the id, and it has to be run on an array.

Reply:
```
return recipe.ingredients.includes(ingredientMatch.id)
```

Reply:
ahh Yes youre right I think theres one more layer I need to filter(?) through because ingredients is an array of objects